### PR TITLE
Sentiment Analysis, continued

### PR DESCRIPTION
**Pull request by aviks**
Supersedes #77 

This PR builds on top of @ayush1999 's work - much appreciated. Changes include:

* Use BSON instead of hdf5 to store weights
* Use BinaryProvider to download and manage model files
   * model files are now stored in a single `.tar.gz` archive

as a result, a `Pkg.build` will download the files, and write a `deps.jl` with the correct paths. 

A further change might be to read the weights once, rather than each time, but that can be done later. 

cc: @MikeInnes

### CODE REVIEW

**Comment by aviks** [comment]
@ayush1999 One think i need from you is the location of the original model. We'll need to have a link to it in the docs. 

@MikeInnes would appreciate a review from you on the final form of this PR.

**Comment by ayush1999** [comment]
@aviks The original Keras model code can be found [here](https://github.com/ayush1999/Keras.jl/blob/master/examples/imdb%20sentiment%20analysis/main.py). The sentiment analysis model in this PR is just the Flux version of this keras model.
Also, I've made a small change in the behavior of the `Flatten` function (though it doesn't make a lot of difference, but it's better to maintain uniformity) in my previous PR, [here](https://github.com/ayush1999/TextAnalysis.jl/commit/276368b3f627e49b85f82cfbb4bacf64085f909e). Can you please include that too?

Please tell me if you'd need any other details.

**Comment by aviks** [comment]
@ayush1999 that link goes to `Keras.jl` in your github account. Can you point me to where in the keras source this originally came from? Also, where is the weights file originally stored?

File: src/sentiment.jl [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+function read_word_ids(filename=sentiment_words)
+    return JSON.parse(String(read(open(filename, "r"))))
+end
+
+function embedding(embedding_matrix, x)
+    temp = embedding_matrix[:, Int64(x[1])+1]
+    for i=2:length(x)
+        temp = hcat(temp, embedding_matrix[:, Int64(x[i])+1])
+    end
+    return reshape(temp, reverse(size(temp)))
```
Comment (MikeInnes) [inline src/sentiment.jl]: This seems odd, can't you just use indexing here? What's the reshape for?

File: src/sentiment.jl [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+end
+
+function flatten(x)
+    return vec(x)
```
Comment (MikeInnes) [inline src/sentiment.jl]: Perhaps just use `vec` directly?

File: src/sentiment.jl [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+end
+
+function get_sentiment(ip::Array{T, 1}, weight, rwi) where T <: AbstractString
+    model = (x,) -> begin
+    a_1 = embedding(weight[:embedding_1]["embedding_1"]["embeddings:0"], x)
+    a_2 = flatten(a_1)
+    a_3 = Dense(weight[:dense_1]["dense_1"]["kernel:0"], weight[:dense_1]["dense_1"]["bias:0"], relu)(a_2)
+    a_4 = Dense(weight[:dense_2]["dense_2"]["kernel:0"], weight[:dense_2]["dense_2"]["bias:0"], sigmoid)(a_3)
```
Comment (MikeInnes) [inline src/sentiment.jl]: This model structure should be a `Chain` that's stored in the `SentimentAnalyser` struct.

**Comment by aviks** [comment]
@ayush1999 where do the weights come from originally? Or did you do the training yourself?

**Pull request merged**